### PR TITLE
🐛  MySQL CDC: Update CDC timestamp fields to be timestamp not ms

### PR DESCRIPTION
**Pull request by danieldiamond**
## What
Changing the default debezium timestamp to be timestamp not milliseconds.

Resolving this issue: https://github.com/airbytehq/airbyte/issues/4159

## How
Add a `transactionTimestamp` field to be used instead of the current `transactionMillis`

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by sherifnada — COMMENTED** [review]
@danieldiamond a few questions (this is my first time looking at this issue): 

1. What's the inconvenience with using the integer timestamp and potentially casting it as needed? 
2. Can we add this to a new field that is just postfixed with `_timestamp`? This removes the need for current users to run a migration or reset their connections
3. since we're using [json-schema](json-schema.org) we should set `type: string` and `format:` it according to the date formats [here](https://json-schema.org/understanding-json-schema/reference/string.html#dates-and-times)

**Comment by danieldiamond** [comment]
@sherifnada thanks for the questions!

1. We have a large dbt project that is configured to run in both redshift & snowflake - this is implies that the raw source data being migrated to both warehouses with the same schema. It simply comes down to the fact that the CDC data is not the same schema as the source data (or the non-CDC syncing method). The alternative suggestion could've been to reconfigure _all_ our dbt staging tables with a conditional `{% if snowflake cast ... %}` (a hacky and time consuming task which I'd be fine to do) however, the issue is that we perform an extensive amount of schema tests on the `source` data, and thus require the actual source table to be the same schema.
2. I'd argue that the other way around would be more appropriate i.e. postfixed with `_ms` as this is an "internal" debezium-related effect. I completely appreciate the desire to avoid current users needing to run migration or reset connection but
    - it's probably a worthwhile effort to do it earlier rather than later, especially as proposed changes are expected behaviour instead of all new users running into these issues moving forward
    - I'd also be curious how many current users are using CDC, based on feedback in slack channel i feel like adoption rate is currently low and of those users, how many are currently doing this manual recasting that would prefer this change
3. can do, but lets first discuss further and decide how to proceed

tl;dr 

* what i'm proposing is expected behaviour. current behaviour cases users to perform a workaround. on the topic of avoiding existing users needing to migrate, what about all the millions of future users we expect 🚀 !!!
* the sooner we do this the better? (fewer current users)

**Comment by marcosmarxm** [comment]
@Phlair and @subodh1810 can you see this change and validate that there is no issue with Debezium internals?

**Comment by danieldiamond** [comment]
> since we're using json-schema we should set type: string and format: it according to the date formats here

Unable to figure out how to include `format` into the `Jsons.jsonNode(ImmutableMap.of("type", "string"));`

**Comment by subodh1810** [comment]
Hey @danieldiamond I will work tomorrow on getting this across the finish line. Thank you for contributing :D

**Comment by subodh1810** [comment]
Closing cause these changes are handled in the PR https://github.com/airbytehq/airbyte/pull/5356
